pkg/build/builder/util.go

```diff
@@ -24,5 +24,11 @@ func getBuildEnvVars(build *buildapi.Build) map[string]string {
 		build.Parameters.Revision.Git.Commit != "" {
 		envVars["OPENSHIFT_BUILD_COMMIT"] = build.Parameters.Revision.Git.Commit
 	}
+	if build.Parameters.Strategy.Type == buildapi.STIBuildStrategyType {
```

Comment:
yes it allows them to be overridden, that was by design because I think it's always useful to give user's the ability to override our internal behavior as a potential workaround if we missed a use case.  The likelihood of them accidentally overriding it seems low.

As for using the mergeWithoutDuplicates, that operates on a different set of types than we have here (requires both args by kapi.EnvVar)